controller/kubernetes_secret_controller.go

```diff
@@ -199,3 +191,46 @@ func (ks *KubernetesSecretController) enqueueSecretChange(obj interface{}) {
 
 	ks.queue.AddRateLimited(key)
 }
+
+// annotateAWSIAMRoleArn ensures that the running pods of the manager as well as the replica instance managers.
+// have the correct AWS IAM role arn assigned to them based on the passed `awsIAMRoleArn`
+func (ks *KubernetesSecretController) annotateAWSIAMRoleArn(awsIAMRoleArn string) error {
+	managerPods, err := ks.ds.ListManagerPods()
+	if err != nil {
+		return err
+	}
+
+	imPods, err := ks.ds.ListInstanceManagerPodsBy(ks.controllerID, "", types.InstanceManagerTypeReplica)
+	if err != nil {
+		return err
+	}
+
+	pods := append(managerPods, imPods...)
+	for _, pod := range pods {
+		if pod.Spec.NodeName != ks.controllerID {
+			continue
+		}
```

Comment:
This is what you implemented to base on the `current ownership model` to handle the resources belonging to it. So it's also for manager pod not only instance manager pod.

Reply:
okay!

Reply:
If we would be able to retrieve just a subset for the manager pods (i.e the ones running on this node), this wouldn't be needed.
Currently there is no label on the daemonset spawned manager pods that would allow us to do so.